### DISCUSSION TITLE
Missing content-length header on audio file

### DISCUSSION THREAD

**Discussion by NilsChr** [pocketbase/pocketbase#5058] [Q&A] [answered]
I'm trying to create an audio player which fetches mp3 files from my pocketbase instance. 
I noticed that the duration of the audio is always "Infinite" or undefined. 
This causes song players to not work properly and treats the audio as a "Broadcast" instead of regular audio. 

I did some testing towards an external site and i can see that these headers are missing from pocketbase (values are examples): 
```
Content-Length: 7807647
Content-Range:bytes 2818048-10625694/10625695
Content-Type:audio/mpeg
```
How can this be added to files when they are downloaded?

**Comment by ganigeorgiev** [comment, marked as answer]
I'm not able to reproduce it. From the live demo (_the type of the file shouldn't matter_):

```
https://pocketbase.io/api/files/POWMOh0W6IoLUAI/eP2jCr1h3NGtsbz/VPtmjtDBX6noQL6iSg6CEvYVhcaa1uAw.jpg

Accept-Ranges: bytes
Cache-Control: max-age=2592000, stale-while-revalidate=86400
Content-Disposition: inline; filename=VPtmjtDBX6noQL6iSg6CEvYVhcaa1uAw.jpg
Content-Length: 517824
Content-Security-Policy: default-src 'none'; media-src 'self'; style-src 'unsafe-inline'; sandbox
Content-Type: image/jpeg
Date: Sun, 09 Jun 2024 10:09:33 GMT
Last-Modified: Sat, 29 Oct 2022 22:22:15 GMT
Vary: Origin
X-Content-Type-Options: nosniff
X-Xss-Protection: 1; mode=block
```

The missing headers could be caused by a misconfigured reverse proxy, I'm not sure.

**Comment by NilsChr** [comment]
Thanks for the super quick response! 
I did another test now and uploaded via pockethost which works. 

The setup i have which is not working correctly is a self hosted Coolify instance on an ubuntu server. 
Will write back if I find some fix for this for others in the future.

**Reply by dosraytech** [reply]
@NilsChr Have you found any solution to this yet? Would love to hear if you’ve had any luck! I faced this issue too. Thanks

**Reply by ganigeorgiev** [reply]
@dosraytech You are commenting on a very old discussion without providing details about your environment setup. As mentioned above the issue is not PocketBase related and could be caused by a reverse-proxy (_in the author's case I guess Caddy_) or some other env misconfiguration.

If you are still not able to identify the culprit, please open a new Q&A discussion with details about your environment to attempt to reproduce it.

**Comment by godmar** [comment]
> I'm trying to create an audio player which fetches mp3 files from my pocketbase instance. I noticed that the duration of the audio is always "Infinite" or undefined. This causes song players to not work properly and treats the audio as a "Broadcast" instead of regular audio.
> 
> I did some testing towards an external site and i can see that these headers are missing from pocketbase (values are examples):
> 
> ```
> Content-Length: 7807647
> Content-Range:bytes 2818048-10625694/10625695
> Content-Type:audio/mpeg
> ```
> 
> How can this be added to files when they are downloaded?

BTW, your example is *not* a range request (or at least it's not clear that you issued a range request); the Content-Range header is included only in responses to range requests.
But the demo supports those too if you use for instance curl's `--range` option to make a range request:

```
$ curl --range 0-9 -v https://pocketbase.io/api/files/POWMOh0W6IoLUAI/eP2jCr1h3NGtsbz/VPtmjtDBX6noQL6iSg6CEvYVhcaa1uAw.jpg |& grep -i 'content'
< content-disposition: inline; filename=VPtmjtDBX6noQL6iSg6CEvYVhcaa1uAw.jpg
< content-range: bytes 0-9/517824
< content-security-policy: default-src 'none'; media-src 'self'; style-src 'unsafe-inline'; sandbox
< content-type: image/jpeg
< x-content-type-options: nosniff
< content-length: 10
```
so things appear to work as advertised.  Check your client request.
